### PR TITLE
rework namespaces to reduce duplication and typing

### PR DESCRIPTION
**Pull request by e0ff**

### CODE REVIEW

File: src/auth.zig [hunk 1]
```diff
@@ -1,10 +1,6 @@
 pub const Auth = @This();
 
 pub const Provider = @import("auth/provider.zig");
-
-pub const AuthZ = @import("authorization.zig");
```
Comment (GrayHatter) [inline src/auth.zig:5]: if you want to drop these, delete the files as well

File: src/endpoint.zig [hunk 2]
```diff
@@ -68,7 +68,9 @@ fn buildRoutes(EP: anytype) [routeCount(EP)]Router.Match {
 
 const std = @import("std");
 const Allocator = std.mem.Allocator;
-const Verse = @import("verse.zig");
+const router = @import("router.zig");
+const server = @import("server.zig");
```
Comment (GrayHatter) [inline src/endpoint.zig]:
```suggestion
const Server = @import("server.zig");
```
it's a type with fields

File: src/endpoint.zig [hunk 3]
```diff
@@ -68,7 +68,9 @@ fn buildRoutes(EP: anytype) [routeCount(EP)]Router.Match {
 
 const std = @import("std");
 const Allocator = std.mem.Allocator;
-const Verse = @import("verse.zig");
+const router = @import("router.zig");
```
Comment (GrayHatter) [inline src/endpoint.zig]:
```suggestion
const Router = @import("router.zig");
```

File: src/verse.zig [hunk 4]
```diff
@@ -1,366 +1,49 @@
...
-const eql = std.mem.eql;
-const Stream = std.net.Stream;
-const AnyWriter = std.io.AnyWriter;
-const bufPrint = std.fmt.bufPrint;
-const allocPrint = std.fmt.allocPrint;
-const splitScalar = std.mem.splitScalar;
-const log = std.log.scoped(.Verse);
-const iovec = std.posix.iovec;
-const iovec_c = std.posix.iovec_const;
+const frame = @import("frame.zig");
```
Comment (GrayHatter) [inline src/verse.zig]: sorry I'm gonna be pendantic about this (I want to preserve as much as the git blame as I can)

Lets rename this file to root.zig

and then in a 2nd commit rename it back to verse.zig, that way git will create a new tree for it

Comment (GrayHatter) [reply src/verse.zig]: and then most of these should be TitleCase

**Review by GrayHatter — CHANGES_REQUESTED** [review]

File: src/verse.zig [hunk 5]
```diff
@@ -1,366 +1,49 @@
...
+pub const DataItem = request_data.DataItem;
+pub const Validator = request_data.Validator;
+
+pub const Cookie = cookies.Cookie;
+
+pub const ContentType = content_type.ContentType;
+
+pub const Template = template.Template;
+pub const PageData = template.PageData;
+pub const findTemplate = template.findTemplate;
```
Comment (GrayHatter) [inline src/verse.zig]: Don't use multiple lines when you only need to use one. 

The use of `pub const Name = @This();` when it a Type with fields is for convience inside the function. I'd rather drop `pub` from `pub const` than use these extra definitions;

File: src/verse.zig [hunk 6]
```diff
@@ -1,366 +1,49 @@
...
+pub const Endpoints = endpoint.Endpoints;
+
+pub const RoutingError = router.RoutingError;
+pub const ServerError = errors.ServerError;
+pub const ClientError = errors.ClientError;
+pub const NetworkError = errors.NetworkError;
+pub const Error = errors.Error;
+
+pub const html = @import("html.zig");
+pub const ini = @import("ini.zig");
```
Comment (GrayHatter) [inline src/verse.zig:36]: unrelated to this PR, we might want to drop this from verse?

Comment (e0ff) [reply src/verse.zig:36]: I'll do that in a separate pr.

**Pull request merged**